roscala/src/main/scala/coop/rchain/roscala/pools/StrandPoolExecutor.scala

```diff
@@ -19,31 +19,34 @@ object StrandPoolExecutor {
 
   def instance[E: StrandPoolExecutor](implicit ev: StrandPoolExecutor[E]): ev.Tpe = ev.instance
 
-  implicit val parallelExecutor = new StrandPoolExecutor[ParallelStrandPool] {
-    private val pool     = new ParallelStrandPool
+  implicit def parallelExecutor = new StrandPoolExecutor[ParallelStrandPool] {
+    private val tracker  = new Phaser(1)
```

Comment:
That is because parallel sp's main thread will arrive twice: first in `StrandPoolExecutor`s start method and second in `Vm::run` with `ParallelStrandPool::finish`.